controller/readwrite.go

```diff
@@ -373,30 +373,35 @@ func (rw *ReadWrite) createTask(ctx context.Context, taskConfig config.TaskConfi
 	d, err := rw.createNewTaskDriver(taskConfig)
 	if err != nil {
 		logger.Error("error creating new task driver", "error", err)
-		return nil, fmt.Errorf("error creating new task driver: %v", err)
+		return nil, err
 	}
 
 	// Initialize the new task
 	err = d.InitTask(ctx)
 	if err != nil {
-		return nil, fmt.Errorf("error initializing new task, %s", err)
+		logger.Error("error initializing new task", "error", err)
+		return nil, err
 	}
 
+	timeout := time.After(5 * time.Minute)
 	for {
 		select {
 		case <-ctx.Done():
 			return nil, ctx.Err()
+		case <-timeout:
+			return nil, fmt.Errorf("timed out rendering template for task '%s'", taskName)
 		default:
 		}
 		ok, err := d.RenderTemplate(ctx)
 		if err != nil {
 			logger.Error("error rendering task template")
-			return nil, fmt.Errorf("error rendering template for task '%s': %s", taskName, err)
+			return nil, err
 		}
 		if ok {
 			// Once template rendering is finished, return
 			return d, nil
 		}
+		time.Sleep(250 * time.Millisecond) // waiting because cannot block on a dependency change
```

Comment:
I know a sleep is not a great solution, but the constant logging was really ruining our logs in TRACE mode and I wanted to put in at least a stop gap for now. Happy to discuss other options or to adjust the sleep time to lower!

Reply:
yeah, I'm okay with this for now, Melissa and I discussed async that we should revisit logs in general and perhaps do an audit. As it stands right now, trace is the necessary log level for development work, but if most of the development level logs were debug logs instead, then high output trace logs like this would be more acceptable.

Reply:
Decreased down to 50 ms.